card/db/migrate_core_cards/20150429090551_search_card_context.rb

```diff
@@ -0,0 +1,25 @@
+# -*- encoding : utf-8 -*-
+
+class SearchCardContext < Card::CoreMigration
+  def up
+    sep = %r{\W}
+    replace = [
+      ['[lr]+','l\\1'],
+      ['[LR]+','L\\1'],
+      ["(?=[lrLR]+#{sep})(?=[LR]*[lr]+)(?=[lr]*[LR]+).*",'l\\1'],   # mix of lowercase and uppercase l's and r's
+      ['left',   'LL'],
+      ['right',  'LR'],
+      ['self',   'left'],
+      ['',       'left'],
+    ]
+    Card.search(:type=>'search').each do |card|
```

Comment:
ok, I'll change that. When I wrote that I wasn't sure if ID uses the cardname or the codename, if it's SearchID or SearchTypeID so I used type instead of type_id.